<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="robots" content="noodp" />
        <title>【bioinfo】-Q&amp;A-42 - Milky板板の博客</title><meta name="Description" content="Milky板板"><meta property="og:title" content="【bioinfo】-Q&amp;A-42" />
<meta property="og:description" content="用米氏方程解决单细胞转录组dropout现象  米氏方程（Michaelis-Menten equation）: v=Vmax × [S] /（Km&#43;[S]) 在假定存在一个稳态反应条件下推导出来的，其中 Km 值称为米氏常数，Vmax是酶被底物饱和时的反应速度，[S]为底物浓度。 Km值的物理意义为反应速度（v）达到1/2Vmax时的底物浓度（即Km=[S]），单位一般为mol/L，只由酶的性质决定，而与酶的浓度无关。可用Km的值鉴别不同的酶。
 今天要介绍的这篇文章提出了一个算法，R包是：M3Drop ， 文章是：Modelling dropouts for feature selection in scRNASeq experiments
挑选重要基因 目前已有的寻找单细胞转录组测序数据中的重要基因（feature selection）的方法都不够好，比如 scLVM 主要是根据先验基因集，比如cell-cycle or apoptosis来区分细胞。与此相反，基于 highly variable genes (HVG) 的方法挑选到的变化量大的那些基因很可能是技术带来的误差。而且低表达量基因的变动往往大于高表达量基因，而且所谓的表达变化大也并没有很好的生物学解释。 一个比较好理解的概念是差异基因，但是需要预先把细胞群体分组后进行比较才能得到，而很多时候细胞太相似了，没办法很好的分开。像PCA或者t-SNE这样的降维方法也可以用来挑选重要基因，但它们也受制于系统误差或者批次误差等等。 dropout是scRNASeq数据的一大特点，就是很多基因在某些细胞根本就不表达，但是在另外的细胞却高表达。这篇文章作者对全长转录本数据和基于UMI的表达量数据分别提出了对应的解决方案，Michaelis-Menten equation 和 depth adjusted negative binomial (DANB) 。
单细胞转录组数据里面的dropouts可以达到50%，但是通常认为这个dropouts是因为在文库构建的过程中，有部分基因没有被成功的反转录，是一个酶促反应。 所以作者用Michaelis-Menten 来建模。
比较9种 feature selection 方法 使用它们分别对基因排序，算法如下：
1 2 3 4 5 6 7 8  bythemagnitudeoftheirloadingsinprincipalcomponentanalysis(PCA)bythestrengthoftheirmostnegativegene-genecorrelation(Cor)bytheirrelativeGiniindex(Gini)M3Dropdropouts-meanexpressioncurve(M3Drop)thesquaredcoefficientofvariation(CV2)meanexpressionrelationship(HVG)thedispersion-meanexpressionrelationshipfitbyDANB(NBDisp)thedropouts-meanexpressionrelationshipfitbyDANB(NBDrop).  这些算法都不需要预先对样本进行分类，是无监督的算法。
1 2 3  differentially variable (DV)genes highly variable (HV) genes differentially expressed (DE) genes   单细胞转录组数据的batch effects比较严重，所以 feature selection 过程的一个主要目的就是降低技术误差的影响，集中在有生物学意义的差异上面。" />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://uioqps.github.io/bioinfo-q-a-42/" /><meta property="og:image" content="https://uioqps.github.io/logo.png"/><meta property="article:section" content="posts" />
<meta property="article:published_time" content="2020-02-20T19:42:22+00:00" />
<meta property="article:modified_time" content="2020-02-20T19:42:22+00:00" /><meta property="og:site_name" content="Milky板板の博客" />

<meta name="twitter:card" content="summary_large_image"/>
<meta name="twitter:image" content="https://uioqps.github.io/logo.png"/>

<meta name="twitter:title" content="【bioinfo】-Q&amp;A-42"/>
<meta name="twitter:description" content="用米氏方程解决单细胞转录组dropout现象  米氏方程（Michaelis-Menten equation）: v=Vmax × [S] /（Km&#43;[S]) 在假定存在一个稳态反应条件下推导出来的，其中 Km 值称为米氏常数，Vmax是酶被底物饱和时的反应速度，[S]为底物浓度。 Km值的物理意义为反应速度（v）达到1/2Vmax时的底物浓度（即Km=[S]），单位一般为mol/L，只由酶的性质决定，而与酶的浓度无关。可用Km的值鉴别不同的酶。
 今天要介绍的这篇文章提出了一个算法，R包是：M3Drop ， 文章是：Modelling dropouts for feature selection in scRNASeq experiments
挑选重要基因 目前已有的寻找单细胞转录组测序数据中的重要基因（feature selection）的方法都不够好，比如 scLVM 主要是根据先验基因集，比如cell-cycle or apoptosis来区分细胞。与此相反，基于 highly variable genes (HVG) 的方法挑选到的变化量大的那些基因很可能是技术带来的误差。而且低表达量基因的变动往往大于高表达量基因，而且所谓的表达变化大也并没有很好的生物学解释。 一个比较好理解的概念是差异基因，但是需要预先把细胞群体分组后进行比较才能得到，而很多时候细胞太相似了，没办法很好的分开。像PCA或者t-SNE这样的降维方法也可以用来挑选重要基因，但它们也受制于系统误差或者批次误差等等。 dropout是scRNASeq数据的一大特点，就是很多基因在某些细胞根本就不表达，但是在另外的细胞却高表达。这篇文章作者对全长转录本数据和基于UMI的表达量数据分别提出了对应的解决方案，Michaelis-Menten equation 和 depth adjusted negative binomial (DANB) 。
单细胞转录组数据里面的dropouts可以达到50%，但是通常认为这个dropouts是因为在文库构建的过程中，有部分基因没有被成功的反转录，是一个酶促反应。 所以作者用Michaelis-Menten 来建模。
比较9种 feature selection 方法 使用它们分别对基因排序，算法如下：
1 2 3 4 5 6 7 8  bythemagnitudeoftheirloadingsinprincipalcomponentanalysis(PCA)bythestrengthoftheirmostnegativegene-genecorrelation(Cor)bytheirrelativeGiniindex(Gini)M3Dropdropouts-meanexpressioncurve(M3Drop)thesquaredcoefficientofvariation(CV2)meanexpressionrelationship(HVG)thedispersion-meanexpressionrelationshipfitbyDANB(NBDisp)thedropouts-meanexpressionrelationshipfitbyDANB(NBDrop).  这些算法都不需要预先对样本进行分类，是无监督的算法。
1 2 3  differentially variable (DV)genes highly variable (HV) genes differentially expressed (DE) genes   单细胞转录组数据的batch effects比较严重，所以 feature selection 过程的一个主要目的就是降低技术误差的影响，集中在有生物学意义的差异上面。"/>
<meta name="application-name" content="Milky板板の博客">
<meta name="apple-mobile-web-app-title" content="Milky板板の博客"><link rel="shortcut icon" type="image/x-icon" href="/favicon.ico" />
        <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
        <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="manifest" href="/site.webmanifest"><link rel="canonical" href="https://uioqps.github.io/bioinfo-q-a-42/" /><link rel="prev" href="https://uioqps.github.io/bioinfo-q-a-41/" /><link rel="next" href="https://uioqps.github.io/bioinfo-q-a-43/" /><link rel="stylesheet" href="/css/style.min.css"><link rel="preload" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@6.1.1/css/all.min.css" as="style" onload="this.onload=null;this.rel='stylesheet'">
        <noscript><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@6.1.1/css/all.min.css"></noscript><link rel="preload" href="https://cdn.jsdelivr.net/npm/animate.css@4.1.1/animate.min.css" as="style" onload="this.onload=null;this.rel='stylesheet'">
        <noscript><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/animate.css@4.1.1/animate.min.css"></noscript><script type="application/ld+json">
    {
        "@context": "http://schema.org",
        "@type": "BlogPosting",
        "headline": "【bioinfo】-Q\u0026A-42",
        "inLanguage": "en",
        "mainEntityOfPage": {
            "@type": "WebPage",
            "@id": "https:\/\/uioqps.github.io\/bioinfo-q-a-42\/"
        },"image": [{
                            "@type": "ImageObject",
                            "url": "https:\/\/uioqps.github.io\/images\/Apple-Devices-Preview.png",
                            "width":  3200 ,
                            "height":  2048 
                        }],"genre": "posts","keywords": "AQ100, dropout, sc","wordcount":  200 ,
        "url": "https:\/\/uioqps.github.io\/bioinfo-q-a-42\/","datePublished": "2020-02-20T19:42:22+00:00","dateModified": "2020-02-20T19:42:22+00:00","license": "This work is licensed under a Creative Commons Attribution-NonCommercial 4.0 International License.","publisher": {
            "@type": "Organization",
            "name": "xxxx","logo": {
                    "@type": "ImageObject",
                    "url": "https:\/\/uioqps.github.io\/images\/avatar.png",
                    "width":  588 ,
                    "height":  614 
                }},"author": {
                "@type": "Person",
                "name": "Milky"
            },"description": ""
    }
    </script></head>
    <body data-header-desktop="fixed" data-header-mobile="auto"><script type="text/javascript">(window.localStorage && localStorage.getItem('theme') ? localStorage.getItem('theme') === 'dark' : ('auto' === 'auto' ? window.matchMedia('(prefers-color-scheme: dark)').matches : 'auto' === 'dark')) && document.body.setAttribute('theme', 'dark');</script>

        <div id="mask"></div><div class="wrapper"><header class="desktop" id="header-desktop">
    <div class="header-wrapper">
        <div class="header-title">
            <a href="/" title="Milky板板の博客"><span class="header-title-pre"><i class='fa-sharp fa-light fa-dolphin' aria-hidden='true'></i></span>Milky板板の博客</a>
        </div>
        <div class="menu">
            <div class="menu-inner"><a class="menu-item" href="/posts/"> Posts </a><a class="menu-item" href="/tags/"> Tags </a><a class="menu-item" href="/categories/"> Categories </a><a class="menu-item" href="/categories/documentation/"> Docs </a><a class="menu-item" href="/about/"> About </a><a class="menu-item" href="https://github.com/uioQPS/" title="GitHub" rel="noopener noreffer" target="_blank"><i class='fab fa-github fa-fw' aria-hidden='true'></i>  </a><span class="menu-item delimiter"></span><span class="menu-item search" id="search-desktop">
                        <input type="text" placeholder="Search titles or contents..." id="search-input-desktop">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-desktop" title="Search">
                            <i class="fas fa-search fa-fw" aria-hidden="true"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-desktop" title="Clear">
                            <i class="fas fa-times-circle fa-fw" aria-hidden="true"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-desktop">
                            <i class="fas fa-spinner fa-fw fa-spin" aria-hidden="true"></i>
                        </span>
                    </span><a href="javascript:void(0);" class="menu-item theme-switch" title="Switch Theme">
                    <i class="fas fa-adjust fa-fw" aria-hidden="true"></i>
                </a></div>
        </div>
    </div>
</header><header class="mobile" id="header-mobile">
    <div class="header-container">
        <div class="header-wrapper">
            <div class="header-title">
                <a href="/" title="Milky板板の博客"><span class="header-title-pre"><i class='fa-sharp fa-light fa-dolphin' aria-hidden='true'></i></span>Milky板板の博客</a>
            </div>
            <div class="menu-toggle" id="menu-toggle-mobile">
                <span></span><span></span><span></span>
            </div>
        </div>
        <div class="menu" id="menu-mobile"><div class="search-wrapper">
                    <div class="search mobile" id="search-mobile">
                        <input type="text" placeholder="Search titles or contents..." id="search-input-mobile">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-mobile" title="Search">
                            <i class="fas fa-search fa-fw" aria-hidden="true"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-mobile" title="Clear">
                            <i class="fas fa-times-circle fa-fw" aria-hidden="true"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-mobile">
                            <i class="fas fa-spinner fa-fw fa-spin" aria-hidden="true"></i>
                        </span>
                    </div>
                    <a href="javascript:void(0);" class="search-cancel" id="search-cancel-mobile">
                        Cancel
                    </a>
                </div><a class="menu-item" href="/posts/" title="">Posts</a><a class="menu-item" href="/tags/" title="">Tags</a><a class="menu-item" href="/categories/" title="">Categories</a><a class="menu-item" href="/categories/documentation/" title="">Docs</a><a class="menu-item" href="/about/" title="">About</a><a class="menu-item" href="https://github.com/uioQPS/" title="GitHub" rel="noopener noreffer" target="_blank"><i class='fab fa-github fa-fw' aria-hidden='true'></i></a><a href="javascript:void(0);" class="menu-item theme-switch" title="Switch Theme">
                <i class="fas fa-adjust fa-fw" aria-hidden="true"></i>
            </a></div>
    </div>
</header><div class="search-dropdown desktop">
        <div id="search-dropdown-desktop"></div>
    </div>
    <div class="search-dropdown mobile">
        <div id="search-dropdown-mobile"></div>
    </div><main class="main">
                <div class="container"><div class="toc" id="toc-auto">
            <h2 class="toc-title">Contents</h2>
            <div class="toc-content" id="toc-content-auto"></div>
        </div><article class="page single"><h1 class="single-title animate__animated animate__flipInX">【bioinfo】-Q&A-42</h1><div class="post-meta">
            <div class="post-meta-line"><span class="post-author"><a href="https://github.com/uioQPS/" title="Author" target="_blank" rel="noopener noreffer author" class="author"><i class="fas fa-user-circle fa-fw" aria-hidden="true"></i>Milky</a></span>&nbsp;<span class="post-category">included in <a href="/categories/bioinfo/"><i class="far fa-folder fa-fw" aria-hidden="true"></i>bioinfo</a>&nbsp;<a href="/categories/ngs/"><i class="far fa-folder fa-fw" aria-hidden="true"></i>ngs</a></span></div>
            <div class="post-meta-line"><i class="far fa-calendar-alt fa-fw" aria-hidden="true"></i>&nbsp;<time datetime="2020-02-20">2020-02-20</time>&nbsp;<i class="fas fa-pencil-alt fa-fw" aria-hidden="true"></i>&nbsp;200 words&nbsp;
                <i class="far fa-clock fa-fw" aria-hidden="true"></i>&nbsp;One minute&nbsp;<span id="/bioinfo-q-a-42/" class="leancloud_visitors" data-flag-title="【bioinfo】-Q&amp;A-42">
                        <i class="far fa-eye fa-fw" aria-hidden="true"></i>&nbsp;<span class=leancloud-visitors-count></span>&nbsp;views
                    </span>&nbsp;</div>
        </div><div class="details toc" id="toc-static"  data-kept="">
                <div class="details-summary toc-title">
                    <span>Contents</span>
                    <span><i class="details-icon fas fa-angle-right" aria-hidden="true"></i></span>
                </div>
                <div class="details-content toc-content" id="toc-content-static"><nav id="TableOfContents">
  <ul>
    <li><a href="#用米氏方程解决单细胞转录组dropout现象">用米氏方程解决单细胞转录组dropout现象</a>
      <ul>
        <li><a href="#挑选重要基因">挑选重要基因</a></li>
        <li><a href="#比较9种-feature-selection-方法">比较9种 feature selection 方法</a></li>
        <li><a href="#公共数据集">公共数据集</a></li>
      </ul>
    </li>
  </ul>
</nav></div>
            </div><div class="content" id="content"><h2 id="用米氏方程解决单细胞转录组dropout现象">用米氏方程解决单细胞转录组dropout现象</h2>
<blockquote>
<p>米氏方程（Michaelis-Menten equation）: v=Vmax × [S] /（Km+[S])
在假定存在一个稳态反应条件下推导出来的，其中 Km 值称为米氏常数，Vmax是酶被底物饱和时的反应速度，[S]为底物浓度。
Km值的物理意义为反应速度（v）达到1/2Vmax时的底物浓度（即Km=[S]），单位一般为mol/L，只由酶的性质决定，而与酶的浓度无关。可用Km的值鉴别不同的酶。</p>
</blockquote>
<p>今天要介绍的这篇文章提出了一个算法，R包是：M3Drop ， 文章是：Modelling dropouts for feature selection in scRNASeq experiments</p>
<h3 id="挑选重要基因">挑选重要基因</h3>
<p>目前已有的寻找单细胞转录组测序数据中的重要基因（feature selection）的方法都不够好，比如 scLVM 主要是根据先验基因集，比如cell-cycle or apoptosis来区分细胞。与此相反，基于 highly variable genes (HVG) 的方法挑选到的变化量大的那些基因很可能是技术带来的误差。而且低表达量基因的变动往往大于高表达量基因，而且所谓的表达变化大也并没有很好的生物学解释。 一个比较好理解的概念是差异基因，但是需要预先把细胞群体分组后进行比较才能得到，而很多时候细胞太相似了，没办法很好的分开。像PCA或者t-SNE这样的降维方法也可以用来挑选重要基因，但它们也受制于系统误差或者批次误差等等。 dropout是scRNASeq数据的一大特点，就是很多基因在某些细胞根本就不表达，但是在另外的细胞却高表达。这篇文章作者对全长转录本数据和基于UMI的表达量数据分别提出了对应的解决方案，Michaelis-Menten equation 和 depth adjusted negative binomial (DANB) 。</p>
<p>单细胞转录组数据里面的dropouts可以达到50%，但是通常认为这个dropouts是因为在文库构建的过程中，有部分基因没有被成功的反转录，是一个酶促反应。 所以作者用Michaelis-Menten 来建模。</p>
<h3 id="比较9种-feature-selection-方法">比较9种 feature selection 方法</h3>
<p>使用它们分别对基因排序，算法如下：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-zed" data-lang="zed"><span class="line"><span class="cl"><span class="n">by</span><span class="w"> </span><span class="n">the</span><span class="w"> </span><span class="n">magnitude</span><span class="w"> </span><span class="n">of</span><span class="w"> </span><span class="n">their</span><span class="w"> </span><span class="n">loadings</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">principal</span><span class="w"> </span><span class="n">component</span><span class="w"> </span><span class="n">analysis</span><span class="w"> </span><span class="p">(</span><span class="n">PCA</span><span class="p">)</span><span class="w">
</span></span></span><span class="line"><span class="cl"><span class="w"></span><span class="n">by</span><span class="w"> </span><span class="n">the</span><span class="w"> </span><span class="n">strength</span><span class="w"> </span><span class="n">of</span><span class="w"> </span><span class="n">their</span><span class="w"> </span><span class="n">most</span><span class="w"> </span><span class="n">negative</span><span class="w"> </span><span class="n">gene</span><span class="o">-</span><span class="n">gene</span><span class="w"> </span><span class="n">correlation</span><span class="w"> </span><span class="p">(</span><span class="n">Cor</span><span class="p">)</span><span class="w">
</span></span></span><span class="line"><span class="cl"><span class="w"></span><span class="n">by</span><span class="w"> </span><span class="n">their</span><span class="w"> </span><span class="n">relative</span><span class="w"> </span><span class="n">Gini</span><span class="w"> </span><span class="n">index</span><span class="w"> </span><span class="p">(</span><span class="n">Gini</span><span class="p">)</span><span class="w">
</span></span></span><span class="line"><span class="cl"><span class="w"></span><span class="n">M3Drop</span><span class="w"> </span><span class="n">dropouts</span><span class="o">-</span><span class="n">mean</span><span class="w"> </span><span class="n">expression</span><span class="w"> </span><span class="n">curve</span><span class="w"> </span><span class="p">(</span><span class="n">M3Drop</span><span class="p">)</span><span class="w">
</span></span></span><span class="line"><span class="cl"><span class="w"></span><span class="n">the</span><span class="w"> </span><span class="n">squared</span><span class="w"> </span><span class="n">coefficient</span><span class="w"> </span><span class="n">of</span><span class="w"> </span><span class="n">variation</span><span class="w"> </span><span class="p">(</span><span class="n">CV2</span><span class="p">)</span><span class="w">
</span></span></span><span class="line"><span class="cl"><span class="w"></span><span class="n">mean</span><span class="w"> </span><span class="n">expression</span><span class="w"> </span><span class="n">relationship</span><span class="w"> </span><span class="p">(</span><span class="n">HVG</span><span class="p">)</span><span class="w">
</span></span></span><span class="line"><span class="cl"><span class="w"></span><span class="n">the</span><span class="w"> </span><span class="n">dispersion</span><span class="o">-</span><span class="n">mean</span><span class="w"> </span><span class="n">expression</span><span class="w"> </span><span class="n">relationship</span><span class="w"> </span><span class="n">fit</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="n">DANB</span><span class="w"> </span><span class="p">(</span><span class="n">NBDisp</span><span class="p">)</span><span class="w">
</span></span></span><span class="line"><span class="cl"><span class="w"></span><span class="n">the</span><span class="w"> </span><span class="n">dropouts</span><span class="o">-</span><span class="n">mean</span><span class="w"> </span><span class="n">expression</span><span class="w"> </span><span class="n">relationship</span><span class="w"> </span><span class="n">fit</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="n">DANB</span><span class="w"> </span><span class="p">(</span><span class="n">NBDrop</span><span class="p">).</span><span class="w">
</span></span></span></code></pre></td></tr></table>
</div>
</div><p>这些算法都不需要预先对样本进行分类，是无监督的算法。</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">differentially variable (DV)genes
</span></span><span class="line"><span class="cl">highly variable (HV) genes
</span></span><span class="line"><span class="cl">differentially expressed (DE) genes
</span></span></code></pre></td></tr></table>
</div>
</div><p>单细胞转录组数据的batch effects比较严重，所以 feature selection 过程的一个主要目的就是降低技术误差的影响，集中在有生物学意义的差异上面。</p>
<h3 id="公共数据集">公共数据集</h3>
<p>作者比较了 5个公共数据集，都是小鼠的胚胎细胞，含有17~255个细胞的测序数据，包括zygote to blastocyst.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">Tung et al. (2017) [12] considered iPSCs from three different individuals and performed three replicates of UMI-tagged scRNASeq and three replicates of bulk RNASeq for each. (GSE77288 ).
</span></span><span class="line"><span class="cl">For Kolodziejczyk et al. (2015)，we considered ESCs grown under two conditions: alternative 2i and serum for which there were three replicates of scRNASeq and two replicates of bulk RNASeq.( E-MTAB-2600 )
</span></span></code></pre></td></tr></table>
</div>
</div><p>对bulk转录组数据用了3种方法找差异基因，分别是 DESeq2,edgeR,limma-voom，只有3种方法都是 5% FDR的差异基因才认为是阳性标准基因集，那些3种方法都在 20% FDR的非差异基因认为是阴性金标准。</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">1,915 positives, and 8,398 negatives for the iPSCs
</span></span><span class="line"><span class="cl">709 positives and 11,278 negatives for the ESCs
</span></span><span class="line"><span class="cl">有了这些基因，就可以计算ROC
</span></span></code></pre></td></tr></table>
</div>
</div><p>细胞转录组数据文章一般分成下面两大类：</p>
<p>第一类是：deep sequencing of full-transcripts for a relatively small number of cells</p>
<p>第二类是：high-cell number, low-depth sequencing of 3’ or 5’ ends of transcripts tagged with unique molecular identifiers</p>
</div><div class="post-footer" id="post-footer">
    <div class="post-info">
        <div class="post-info-line">
            <div class="post-info-mod">
                <span>Updated on 2020-02-20</span>
            </div></div>
        <div class="post-info-line">
            <div class="post-info-md"><span>
                            <a class="link-to-markdown" href="/bioinfo-q-a-42/index.md" target="_blank">Read Markdown</a>
                        </span></div>
            <div class="post-info-share">
                <span><a href="javascript:void(0);" title="Share on Twitter" data-sharer="twitter" data-url="https://uioqps.github.io/bioinfo-q-a-42/" data-title="【bioinfo】-Q&amp;A-42" data-hashtags="AQ100,dropout,sc"><i class="fab fa-twitter fa-fw" aria-hidden="true"></i></a><a href="javascript:void(0);" title="Share on Facebook" data-sharer="facebook" data-url="https://uioqps.github.io/bioinfo-q-a-42/" data-hashtag="AQ100"><i class="fab fa-facebook-square fa-fw" aria-hidden="true"></i></a><a href="javascript:void(0);" title="Share on Hacker News" data-sharer="hackernews" data-url="https://uioqps.github.io/bioinfo-q-a-42/" data-title="【bioinfo】-Q&amp;A-42"><i class="fab fa-hacker-news fa-fw" aria-hidden="true"></i></a><a href="javascript:void(0);" title="Share on Line" data-sharer="line" data-url="https://uioqps.github.io/bioinfo-q-a-42/" data-title="【bioinfo】-Q&amp;A-42"><i data-svg-src="https://cdn.jsdelivr.net/npm/simple-icons@7.3.0/icons/line.svg" aria-hidden="true"></i></a><a href="javascript:void(0);" title="Share on 微博" data-sharer="weibo" data-url="https://uioqps.github.io/bioinfo-q-a-42/" data-title="【bioinfo】-Q&amp;A-42"><i class="fab fa-weibo fa-fw" aria-hidden="true"></i></a></span>
            </div>
        </div>
    </div>

    <div class="post-info-more">
        <section class="post-tags"><i class="fas fa-tags fa-fw" aria-hidden="true"></i>&nbsp;<a href="/tags/aq100/">AQ100</a>,&nbsp;<a href="/tags/dropout/">dropout</a>,&nbsp;<a href="/tags/sc/">sc</a></section>
        <section>
            <span><a href="javascript:void(0);" onclick="window.history.back();">Back</a></span>&nbsp;|&nbsp;<span><a href="/">Home</a></span>
        </section>
    </div>

    <div class="post-nav"><a href="/bioinfo-q-a-41/" class="prev" rel="prev" title="【bioinfo】-Q&amp;A-41"><i class="fas fa-angle-left fa-fw" aria-hidden="true"></i>【bioinfo】-Q&A-41</a>
            <a href="/bioinfo-q-a-43/" class="next" rel="next" title="【bioinfo】-Q&amp;A-43">【bioinfo】-Q&A-43<i class="fas fa-angle-right fa-fw" aria-hidden="true"></i></a></div>
</div>
<div id="comments"><div id="valine" class="comment"></div><noscript>
                Please enable JavaScript to view the comments powered by <a href="https://valine.js.org/">Valine</a>.
            </noscript></div></article></div>
            </main><footer class="footer">
        <div class="footer-container"><div class="footer-line" itemscope itemtype="http://schema.org/CreativeWork"><i class="far fa-copyright fa-fw" aria-hidden="true"></i><span itemprop="copyrightYear">2023 - 2024</span><span class="author" itemprop="copyrightHolder">&nbsp;<a href="https://github.com/uioQPS/" target="_blank">Milky</a></span>&nbsp;|&nbsp;<span class="license"><a rel="license external nofollow noopener noreffer" href="https://creativecommons.org/licenses/by-nc/4.0/" target="_blank">CC BY-NC 4.0</a></span></div>
        </div>
    </footer></div>

        <div id="fixed-buttons"><a href="#" id="back-to-top" class="fixed-button" title="Back to Top">
                <i class="fas fa-arrow-up fa-fw" aria-hidden="true"></i>
            </a><a href="#" id="view-comments" class="fixed-button" title="View Comments">
                <i class="fas fa-comment fa-fw" aria-hidden="true"></i>
            </a>
        </div><link rel="stylesheet" href="/lib/valine/valine.min.css"><script type="text/javascript" src="https://cdn.jsdelivr.net/npm/valine@1.5.0/dist/Valine.min.js"></script><script type="text/javascript" src="https://cdn.jsdelivr.net/npm/autocomplete.js@0.38.1/dist/autocomplete.min.js"></script><script type="text/javascript" src="https://cdn.jsdelivr.net/npm/lunr@2.3.9/lunr.min.js"></script><script type="text/javascript" src="https://cdn.jsdelivr.net/npm/lazysizes@5.3.2/lazysizes.min.js"></script><script type="text/javascript" src="https://cdn.jsdelivr.net/npm/clipboard@2.0.11/dist/clipboard.min.js"></script><script type="text/javascript" src="https://cdn.jsdelivr.net/npm/sharer.js@0.5.1/sharer.min.js"></script><script type="text/javascript">window.config={"code":{"copyTitle":"Copy to clipboard","maxShownLines":50},"comment":{"valine":{"appId":"QGzwQXOqs5JOhN4RGPOkR2mR-MdYXbMMI","appKey":"WBmoGyJtbqUswvfLh6L8iEBr","avatar":"mp","el":"#valine","emojiCDN":"https://cdn.jsdelivr.net/npm/emoji-datasource-google@14.0.0/img/google/64/","emojiMaps":{"100":"1f4af.png","alien":"1f47d.png","anger":"1f4a2.png","angry":"1f620.png","anguished":"1f627.png","astonished":"1f632.png","black_heart":"1f5a4.png","blue_heart":"1f499.png","blush":"1f60a.png","bomb":"1f4a3.png","boom":"1f4a5.png","broken_heart":"1f494.png","brown_heart":"1f90e.png","clown_face":"1f921.png","cold_face":"1f976.png","cold_sweat":"1f630.png","confounded":"1f616.png","confused":"1f615.png","cry":"1f622.png","crying_cat_face":"1f63f.png","cupid":"1f498.png","dash":"1f4a8.png","disappointed":"1f61e.png","disappointed_relieved":"1f625.png","dizzy":"1f4ab.png","dizzy_face":"1f635.png","drooling_face":"1f924.png","exploding_head":"1f92f.png","expressionless":"1f611.png","face_vomiting":"1f92e.png","face_with_cowboy_hat":"1f920.png","face_with_hand_over_mouth":"1f92d.png","face_with_head_bandage":"1f915.png","face_with_monocle":"1f9d0.png","face_with_raised_eyebrow":"1f928.png","face_with_rolling_eyes":"1f644.png","face_with_symbols_on_mouth":"1f92c.png","face_with_thermometer":"1f912.png","fearful":"1f628.png","flushed":"1f633.png","frowning":"1f626.png","ghost":"1f47b.png","gift_heart":"1f49d.png","green_heart":"1f49a.png","grimacing":"1f62c.png","grin":"1f601.png","grinning":"1f600.png","hankey":"1f4a9.png","hear_no_evil":"1f649.png","heart":"2764-fe0f.png","heart_decoration":"1f49f.png","heart_eyes":"1f60d.png","heart_eyes_cat":"1f63b.png","heartbeat":"1f493.png","heartpulse":"1f497.png","heavy_heart_exclamation_mark_ornament":"2763-fe0f.png","hole":"1f573-fe0f.png","hot_face":"1f975.png","hugging_face":"1f917.png","hushed":"1f62f.png","imp":"1f47f.png","innocent":"1f607.png","japanese_goblin":"1f47a.png","japanese_ogre":"1f479.png","joy":"1f602.png","joy_cat":"1f639.png","kiss":"1f48b.png","kissing":"1f617.png","kissing_cat":"1f63d.png","kissing_closed_eyes":"1f61a.png","kissing_heart":"1f618.png","kissing_smiling_eyes":"1f619.png","laughing":"1f606.png","left_speech_bubble":"1f5e8-fe0f.png","love_letter":"1f48c.png","lying_face":"1f925.png","mask":"1f637.png","money_mouth_face":"1f911.png","nauseated_face":"1f922.png","nerd_face":"1f913.png","neutral_face":"1f610.png","no_mouth":"1f636.png","open_mouth":"1f62e.png","orange_heart":"1f9e1.png","partying_face":"1f973.png","pensive":"1f614.png","persevere":"1f623.png","pleading_face":"1f97a.png","pouting_cat":"1f63e.png","purple_heart":"1f49c.png","rage":"1f621.png","relaxed":"263a-fe0f.png","relieved":"1f60c.png","revolving_hearts":"1f49e.png","right_anger_bubble":"1f5ef-fe0f.png","robot_face":"1f916.png","rolling_on_the_floor_laughing":"1f923.png","scream":"1f631.png","scream_cat":"1f640.png","see_no_evil":"1f648.png","shushing_face":"1f92b.png","skull":"1f480.png","skull_and_crossbones":"2620-fe0f.png","sleeping":"1f634.png","sleepy":"1f62a.png","slightly_frowning_face":"1f641.png","slightly_smiling_face":"1f642.png","smile":"1f604.png","smile_cat":"1f638.png","smiley":"1f603.png","smiley_cat":"1f63a.png","smiling_face_with_3_hearts":"1f970.png","smiling_imp":"1f608.png","smirk":"1f60f.png","smirk_cat":"1f63c.png","sneezing_face":"1f927.png","sob":"1f62d.png","space_invader":"1f47e.png","sparkling_heart":"1f496.png","speak_no_evil":"1f64a.png","speech_balloon":"1f4ac.png","star-struck":"1f929.png","stuck_out_tongue":"1f61b.png","stuck_out_tongue_closed_eyes":"1f61d.png","stuck_out_tongue_winking_eye":"1f61c.png","sunglasses":"1f60e.png","sweat":"1f613.png","sweat_drops":"1f4a6.png","sweat_smile":"1f605.png","thinking_face":"1f914.png","thought_balloon":"1f4ad.png","tired_face":"1f62b.png","triumph":"1f624.png","two_hearts":"1f495.png","unamused":"1f612.png","upside_down_face":"1f643.png","weary":"1f629.png","white_frowning_face":"2639-fe0f.png","white_heart":"1f90d.png","wink":"1f609.png","woozy_face":"1f974.png","worried":"1f61f.png","yawning_face":"1f971.png","yellow_heart":"1f49b.png","yum":"1f60b.png","zany_face":"1f92a.png","zipper_mouth_face":"1f910.png","zzz":"1f4a4.png"},"enableQQ":false,"highlight":true,"lang":"en","pageSize":10,"placeholder":"Your comment ...","recordIP":true,"serverURLs":"https://leancloud.hugoloveit.com","visitor":true}},"search":{"highlightTag":"em","lunrIndexURL":"/index.json","maxResultLength":10,"noResultsFound":"No results found","snippetLength":30,"type":"lunr"}};</script><script type="text/javascript" src="/js/theme.min.js"></script></body>
</html>
